org.eclipse.m2e.repository/pom.xml

```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+	Copyright (c) 2008, 2022 Sonatype, Inc. and others
+	All rights reserved. This program and the accompanying materials
+	are made available under the terms of the Eclipse Public License 2.0
+	which accompanies this distribution, and is available at
+	https://www.eclipse.org/legal/epl-2.0/
+
+	SPDX-License-Identifier: EPL-2.0
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>org.eclipse.m2e</groupId>
+		<artifactId>m2e-core</artifactId>
+		<version>2.0.0-SNAPSHOT</version>
+	</parent>
+
+	<artifactId>org.eclipse.m2e.repository</artifactId>
+	<packaging>eclipse-repository</packaging>
+
+	<name>Maven Integration for Eclipse Repository</name>
+
+	<profiles>
+		<!-- Eclipse SimRel requires all artifacts to be jar- or pgp-signed. So we sign the (non eclipse) artifacts
+			and update the p2 metadata accordingly -->
+		<profile>
+			<id>eclipse-sign</id>
+			<build>
+				<plugins>
+					<plugin>
+						<groupId>org.eclipse.tycho</groupId>
+						<artifactId>tycho-gpg-plugin</artifactId>
+						<version>${tycho-version}</version>
+						<executions>
+							<execution>
+								<id>pgpsigner</id>
+								<goals>
+									<goal>sign-p2-artifacts</goal>
+								</goals>
+								<configuration>
+									<keyname>${gpg.keyname}</keyname>
+									<skipIfJarsigned>true</skipIfJarsigned>
+								</configuration>
+							</execution>
+						</executions>
+					</plugin>
+					<plugin>
+						<groupId>org.eclipse.tycho</groupId>
+						<artifactId>tycho-p2-repository-plugin</artifactId>
```

Comment:
Does the property [tycho-gpg-plugin:addPublicKeysToArtifacts](https://tycho.eclipseprojects.io/doc/3.0.0/tycho-gpg-plugin/sign-p2-artifacts-mojo.html#addPublicKeysToArtifacts) change the included third-party deps?
If yes we have to keep this.

Reply:
`true` is the default value anyway.
This property just adds an extra property to each artifact, it doesn't change *what* is included.

Reply:
So it does only change the metadata in the repo's `artifacts.xml` and not the jar itself, doesn't it?

At least when I just checked the jar's in the repo build on Jenkins the jars of the pgp-signed maven deps where identical to the jars from Maven-Central, so the answer is probably yes.

When looking at the build artifacts.xml I assume that `addPublicKeysToArtifacts` controls if the `pgp.publicKeys` is added to each pgp signed artifact and `addPublicKeyToRepo` controls if `pgp.publicKeys` is added to the 'root' properties.
Can you tell me what's the advantage of adding the pgp.publicKeys to each artifact instead of just adding it to the repo root? To me it just looks like a lot of redundant data.